CRM/Core/CodeGen/Util/Smarty.php

```diff
@@ -43,23 +43,22 @@ public function getCompileDir() {
    * Create a Smarty instance.
    *
    * @return \Smarty
+   * @throws \SmartyException
    */
-  public function createSmarty() {
-    $base = dirname(dirname(dirname(dirname(__DIR__))));
+  public function createSmarty(): Smarty {
+    $base = dirname(__DIR__, 4);
     $pkgs = file_exists(dirname($base) . "/civicrm-packages") ? dirname($base) . "/civicrm-packages" : "$base/packages";
-
-    require_once 'Smarty/Smarty.class.php';
+    require_once $pkgs . '/smarty3/vendor/autoload.php';
     $smarty = new Smarty();
-    $smarty->template_dir = "$base/xml/templates";
-    $smarty->plugins_dir = ["$pkgs/Smarty/plugins", "$base/CRM/Core/Smarty/plugins"];
-    $smarty->compile_dir = $this->getCompileDir();
-    $smarty->clear_all_cache();
+    $smarty->setTemplateDir("$base/xml/templates");
+    $smarty->setPluginsDir(["$pkgs/Smarty/plugins", "$base/CRM/Core/Smarty/plugins"]);
```

Comment:
This would still be loading smarty2 plugins: `$pks/Smarty/plugins`. I don't see a direct equivalent for smarty3?

Reply:
Ah thanks @demeritcowboy  - I'm still bogged down fighting the message template code -but that is probably the next gremlin planning on assaulting me